src/WebJobs.Script.WebHost/ExtensionsOptionProvider.cs

```diff
@@ -0,0 +1,76 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class ExtensionsOptionProvider : IExtensionsOptionProvider
+    {
+        private readonly List<Type> _extensionOptionTypes;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly Dictionary<string, string> _irregularNamingExtensions = new Dictionary<string, string>()
+        {
+            // the extension section name is taken from the prefix of the Option class name. e.g. KafkaOptions -> kafka.
+            // However, some extensions doesn't follow the convention. Replacing the prefix with correct name.
+            { "eventHub", "eventHubs" },
+            { "queues", "queue" },
+            { "blobs", "blob" }
+        };
+
+        private readonly JsonSerializerSettings _settings = new JsonSerializerSettings
+            {
+                ContractResolver = new ExtensionsOptionContractResolver(),
+                ReferenceLoopHandling = ReferenceLoopHandling.Ignore,
+                DefaultValueHandling = DefaultValueHandling.Include,
+                Formatting = Formatting.Indented
+            };
+
+        public ExtensionsOptionProvider(IServiceProvider serviceProvider, IServiceCollection services)
+        {
+            _serviceProvider = serviceProvider;
+            // TODO : The following code might have better ways, however, I can't find it until now.
+            var extensionOptions = services.Where(p => p.ServiceType.IsGenericType && p.ServiceType.GetGenericTypeDefinition() == typeof(IConfigureOptions<>) && p.ImplementationFactory != null && p.ImplementationFactory.Target.ToString().StartsWith(typeof(WebJobsExtensionBuilderExtensions).FullName)).ToList();
+            _extensionOptionTypes = extensionOptions.Select(p => p.ServiceType.GetGenericArguments()[0]).ToList();
+        }
+
+        public Dictionary<string, JObject> GetExtensionOptions()
+        {
+            var configuration = _serviceProvider.GetService<IConfiguration>();
+            // for each of the extension options types identified, create a bound instance
+            // and format to JObject
+            Dictionary<string, JObject> result = new Dictionary<string, JObject>();
+            foreach (Type optionsType in _extensionOptionTypes)
+            {
+                int idx = optionsType.Name.LastIndexOf("Options");
+                string name = optionsType.Name.Substring(0, 1).ToLower() + optionsType.Name.Substring(1, idx - 1);
+                if (_irregularNamingExtensions.ContainsKey(name))
+                {
+                    name = _irregularNamingExtensions[name];
+                }
+                var options = Activator.CreateInstance(optionsType);
+                var key = ConfigurationSectionNames.JobHost + ConfigurationPath.KeyDelimiter
+                     + "extensions" + ConfigurationPath.KeyDelimiter + name;
+                configuration.Bind(key, options);
```

Comment:
Why are you binding manually like this? In my prototype [here](https://github.com/Azure/azure-functions-host/blob/d7ac580aad39489592f3c64d88d52fb4517d5593/src/WebJobs.Script.WebHost/ExtensionOptionsProvider.cs#L43) I was able to allow the IOptions<> object to do that for me. My testing showed that worked - did you hit an issue? That way I didn't need to know configuration sections or anything. I.e. you shouldn't need _irregularNamingExtensions

Reply:
I started testing with you code.  It creates an instance however, it didn't respect the `host.json` and AppSettings. Looks Default Value. If I change the code like it, at least it works.  However, if there is better way, I'd happy to change it.